capsules/extra/src/atecc508a.rs

```diff
@@ -167,6 +177,13 @@ pub struct Atecc508a<'a> {
     entropy_offset: Cell<usize>,
     entropy_client: OptionalCell<&'a dyn entropy::Client32>,
 
+    digest_client: OptionalCell<&'a dyn digest::ClientDataHash<32>>,
+    digest_buffer: TakeCell<'static, [u8; 64]>,
+    write_len: Cell<usize>,
+    remain_len: Cell<usize>,
+    hash_data: Cell<Option<SubSliceMutImmut<'static, u8>>>,
+    digest_data: Cell<Option<&'static mut [u8; 32]>>,
```

Comment:
I think these should be TakeCell or MapCell. Then the `.take()` on line 462 isn't necessary.